crates/virtio-queue/src/mock.rs

```diff
@@ -371,43 +380,64 @@ impl<'a, M: GuestMemory> MockSplitQueue<'a, M> {
         q
     }
 
-    /// Writes a single descriptor chain to the memory object of the queue, at the beginning of the
+    /// Writes one or more descriptor chains to the memory object of the queue, depending on the
+    /// value of the bool variable one_chain, at the beginning of the
     /// descriptor table, and returns the associated `DescriptorChain` object.
-    pub fn build_desc_chain(&self, descs: &[Descriptor]) -> DescriptorChain<&M> {
-        self.add_desc_chain(descs, 0);
-
+    pub fn build_desc_chain(
```

Comment:
Having bool as a parameter that changes the behavior of the function is an antipattern. There are some more details here about this: https://softwareengineering.stackexchange.com/questions/147977/is-it-wrong-to-use-a-boolean-parameter-to-determine-behavior.

Instead, we should have 2 functions, one that builds just one descriptor, and one that builds multiple as specified in the input. Now this would mean going back to the initial solution you proposed. The problem there was the code duplication. One way of solving this is to have a private helper function that takes care of the common code.